An error occurred while executing the following cell:
------------------
import jax
import jax.numpy as jnp  # JAX NumPy

try:
    from flax import linen as nn  # The Linen API
except ModuleNotFoundError:
    %pip install flax
    from flax import linen as nn  # The Linen API
from flax.training import train_state  # Useful dataclass to keep train state
from flax.core.frozen_dict import FrozenDict
from flax.core import freeze, unfreeze

try:
    import torch
except ModuleNotFoundError:
    %pip install torch
    import torch
from torch.utils import data  # For data

import numpy as np  # Ordinary NumPy

try:
    import optax  # Optimizers
except ModuleNotFoundError:
    %pip install optax
    import optax  # Optimizers

import collections
import re
import random
import os
import requests
import zipfile
import tarfile
import hashlib
import time
import json
import multiprocessing
from typing import Any, Callable, Sequence, Tuple
from functools import partial

import matplotlib.pyplot as plt
import pandas as pd
import math
from IPython import display

rng = jax.random.PRNGKey(0)
!mkdir figures # for saving plots
ModuleDef = Any
------------------

---------------------------------------------------------------------------
ModuleNotFoundError                       Traceback (most recent call last)
/tmp/ipykernel_2416/3764372455.py in <module>
     13 try:
---> 14     import torch
     15 except ModuleNotFoundError:

ModuleNotFoundError: No module named 'torch'

During handling of the above exception, another exception occurred:

ImportError                               Traceback (most recent call last)
/tmp/ipykernel_2416/3764372455.py in <module>
     15 except ModuleNotFoundError:
     16     get_ipython().run_line_magic('pip', 'install torch')
---> 17     import torch
     18 from torch.utils import data  # For data
     19 

~/miniconda3/envs/py37/lib/python3.7/site-packages/torch/__init__.py in <module>
    197     if USE_GLOBAL_DEPS:
    198         _load_global_deps()
--> 199     from torch._C import *  # noqa: F403
    200 
    201 # Appease the type checker; ordinarily this binding is inserted by the

ImportError: /github/home/miniconda3/envs/py37/lib/python3.7/site-packages/torch/lib/libtorch_cpu.so: cannot read file data
ImportError: /github/home/miniconda3/envs/py37/lib/python3.7/site-packages/torch/lib/libtorch_cpu.so: cannot read file data
